Makefile

```diff
@@ -48,21 +48,38 @@ fix: $(FIXERS)
 # END: lint-install ../malcontent
 
 SAMPLES_REPO ?= chainguard-dev/malcontent-samples
-SAMPLES_COMMIT ?= 75ee4b4f3e5c29484811e77ffa9d5d02eb75218f
+SAMPLES_COMMIT ?= f56db86f52ef9b647943d9e3b6652a85ccb68266
 OUT_DIR=out/samples-$(SAMPLES_COMMIT).tmp
 out/samples-$(SAMPLES_COMMIT):
 	mkdir -p out
 	git clone https://github.com/$(SAMPLES_REPO).git $(OUT_DIR)
 	git -C $(OUT_DIR) checkout $(SAMPLES_COMMIT)
-	find $(OUT_DIR) -name "*.xz" -execdir tar xJvf "{}" \;
+	@for file in $$(find $(OUT_DIR) -name "*.xz" -print0 | xargs -0 echo); do \
+    dir=$$(dirname "$$file"); \
+    base=$$(basename "$$file" .xz); \
+    fullpath="$$dir/$$base"; \
+    temp_path="$$fullpath".temp; \
+    xz -dkc "$$file" > "$$temp_path"; \
+    if file "$$temp_path" | grep -q "POSIX tar archive"; then \
+      if [ "$(shell uname)" = "Darwin" ]; then \
+        tar xJvf "$$temp_path" -C $$(dirname "$$temp_path"); \
+      elif [ "$(shell uname)" = "Linux" ]; then \
+        tar xvf "$$temp_path" -C $$(dirname "$$temp_path"); \
```

Comment:
Linux should support tar xJvf too?

Reply:
I was getting `tar: short read` errors when using it for some reason (at least in Wolfi).